### PR TITLE
Add configuration for Intent Detection API

### PR DESCRIPTION
**Pull request by rafax**
Adds site-config configuration for RFC 969 intent detection, making the Intent Detection API endpoint and token configurable without code changes. Additionally, adds an option to hit multiple intent detection backends with the same query.

Previously, URL was hardcoded in code, so if the backend has changed, we had to redeploy sourcegraph.com. 
As we iterate on intent detection, we want to be able to test multiple models in parallel, so this PR adds a setting for `extra` backends - if provided, additional .com -> backend requests will be sent, but the client-initiated request will not wait for those requests.

Closes AI-128.

## Test plan

- tested locally - add 
```
 "cody.serverSideContext": {
      "intentDetectionAPI": {
        "default": {
          "url": "http://35.188.42.13:8000/predict/linearv2"
        },
        "extra": [
          {
            "url": "http://35.188.42.13:8000/predict/linearv2"
          }
        ]
      }
    }
```
to `experimentalFeatures` in dev-private.

### CODE REVIEW

File: cmd/frontend/internal/context/resolvers/context.go [hunk 1]
```diff
@@ -128,20 +132,50 @@ func (r *Resolver) ChatIntent(ctx context.Context, args graphqlbackend.ChatInten
...
+			return
+		}
+		r.logger.Debug("fetched intent from extra backend", log.String("interactionID", args.InteractionID), log.String("backend", (*extraBackend).Url), log.String("query", args.Query), log.String("intent", response.Intent), log.Float64("score", response.Score))
+	})
+	if err != nil {
+		return nil, err
+	}
+	r.logger.Info("detecting intent", log.String("interactionID", args.InteractionID), log.String("query", args.Query), log.String("intent", intentResponse.Intent), log.Float64("score", intentResponse.Score))
+	return &chatIntentResponse{intent: intentResponse.Intent, score: intentResponse.Score}, nil
+}
```
Comment (janhartman) [inline cmd/frontend/internal/context/resolvers/context.go:164]: I assume we're dumping the intent somewhere in BQ in the end? If yes, could you please point me to where that is?

File: schema/site.schema.json [hunk 2]
```diff
@@ -630,6 +630,34 @@
           "description": "When enabled, syntactic indexing jobs will be scheduled for all enabled repos",
           "type": "boolean",
           "default": false
+        },
+        "cody.serverSideContext": {
```
Comment (janhartman) [inline schema/site.schema.json:634]: Is schema.go generated from this?

**Review by janhartman — APPROVED** [review]

File: cmd/frontend/internal/context/resolvers/context.go (hunk 1 above)
Comment (rafax) [reply cmd/frontend/internal/context/resolvers/context.go:163]: It's in Google Cloud Logging, where we can add a sink for BigQuery. https://console.cloud.google.com/logs/query;cursorTimestamp=2024-07-17T09:24:41.733603Z;duration=PT1H?project=sourcegraph-dev

Comment (janhartman) [reply cmd/frontend/internal/context/resolvers/context.go:163]: Nice! How hard is it to get that done? It'd be really valuable to have this data for analyses or training.

Comment (rafax) [reply cmd/frontend/internal/context/resolvers/context.go:163]: It's 5m of click-ops, I just want to wait for the messages to start appearing so I can craft a selective filter.

**Pull request merged**